[contributors.json] Relocation

JonWBedard opened the pull request:
#### b4d627ccdaad767a70451a636990638f5cb239b1
<pre>
[contributors.json] Relocation (Part 6)
<a href="https://bugs.webkit.org/show_bug.cgi?id=229690">https://bugs.webkit.org/show_bug.cgi?id=229690</a>
&ltrdar://problem/82552403&gt

Reviewed by NOBODY (OOPS!).

* WebKitBot/src/Commit.mjs: Load contributors from metadata/contributors.json.
* WebKitBot/src/Contributors.mjs: Load contributors from metadata/contributors.json, parse as list instead of dictionary.
* WebKitBot/tests/Commit.test.mjs: Use name instead of fullName, stop relying on nicks.
</pre>
----------------------------------------------------------------------
#### 89526ada7f7fb9686dd3c56ec67f962068f62e9e
```
[contributors.json] Relocation (Part 5)
https://bugs.webkit.org/show_bug.cgi?id=229690
<rdar://problem/82552403>

Reviewed by NOBODY (OOPS!).

* CISupport/ews-build/steps.py:
(ValidateCommiterAndReviewer): Pull from metadata/contributors.json.
(ValidateCommiterAndReviewer.load_contributors_from_disk): Ditto.
(ValidateCommiterAndReviewer.load_contributors): Parse contributors.json as list instead of dictionary.
* CISupport/ews-build/steps_unittest.py:
```
----------------------------------------------------------------------
#### f59a8eee80e807e62573456cff5c484b09fc4ff9
```
[contributors.json] Relocation (Part 4)
https://bugs.webkit.org/show_bug.cgi?id=229690
<rdar://problem/82552403>

Reviewed by NOBODY (OOPS!).

* committers-autocomplete.js: Use metadata/contributors.json.
(parseCommittersPy): Parse contributors.json as list instead of dictionary.
(isMatch): Strip irc references.
(updateMenu): Ditto.
```
----------------------------------------------------------------------
#### 36b2c4b3d434a9b44e6fa2b563974e275fbe5f9a
```
[contributors.json] Relocation (Part 3)
https://bugs.webkit.org/show_bug.cgi?id=229690
<rdar://problem/82552403>

Reviewed by NOBODY (OOPS!).

* commit-review.md: Link to metadata/contributors.json.
* wp-content/themes/webkit/team.php: Load from metadata/contributors.json,
load list instead of dictionary.
```
----------------------------------------------------------------------
#### 08a02d42d652faeb565c3bb7aba57255c565f1bd
```
[contributors.json] Relocation (Part 2)
https://bugs.webkit.org/show_bug.cgi?id=229690
<rdar://problem/82552403>

Reviewed by NOBODY (OOPS!).

* Scripts/webkitpy/common/config/committers.py:
(Contributor.as_dict): Include name in dictionary.
(CommitterList.load_json): Use list instead of dictionary.
(CommitterList.as_json): Encode as list instead of dictionary.
* metadata/contributors.json: Convert from dictionary to list.
(CommitterList._contributor_list_to_dict): Deleted.
```
----------------------------------------------------------------------
#### 2748b23712120aa86e3d0f989151ef267e062fbe
```
[contributors.json] Relocation (Part 1)
https://bugs.webkit.org/show_bug.cgi?id=229690
<rdar://problem/82552403>

Reviewed by Aakash Jain.

* Scripts/webkitpy/common/config/committers.py:
(CommitterList.load_json): Read from metadata/contributors.json.
(CommitterList.reformat_in_place): Ditto.
* metadata/contributors.json: Copied from Tools/Scripts/webkitpy/common/config/contributors.json.
```

rniwa commented:
We don't accept PR requests on Github as our main repository is still in Subversion. Please post a patch with BSD license on bugs.webkit.org instead.

rniwa commented:
Closing this for now. Please follow instructions on https://webkit.org/contributing-code/ In particular, any patch needs to be posted on bugs.webkit.org

JonWBedard committed:
[contributors.json] Relocation (Part 1)
https://bugs.webkit.org/show_bug.cgi?id=229690
<rdar://problem/82552403>

Reviewed by Aakash Jain.

* Scripts/webkitpy/common/config/committers.py:
(CommitterList.load_json): Read from metadata/contributors.json.
(CommitterList.reformat_in_place): Ditto.
* metadata/contributors.json: Copied from Tools/Scripts/webkitpy/common/config/contributors.json.

JonWBedard committed:
[contributors.json] Relocation (Part 2)
https://bugs.webkit.org/show_bug.cgi?id=229690
<rdar://problem/82552403>

Reviewed by NOBODY (OOPS!).

* Scripts/webkitpy/common/config/committers.py:
(Contributor.as_dict): Include name in dictionary.
(CommitterList.load_json): Use list instead of dictionary.
(CommitterList.as_json): Encode as list instead of dictionary.
* metadata/contributors.json: Convert from dictionary to list.
(CommitterList._contributor_list_to_dict): Deleted.

JonWBedard committed:
[contributors.json] Relocation (Part 3)
https://bugs.webkit.org/show_bug.cgi?id=229690
<rdar://problem/82552403>

Reviewed by NOBODY (OOPS!).

* commit-review.md: Link to metadata/contributors.json.
* wp-content/themes/webkit/team.php: Load from metadata/contributors.json,
load list instead of dictionary.

JonWBedard committed:
[contributors.json] Relocation (Part 4)
https://bugs.webkit.org/show_bug.cgi?id=229690
<rdar://problem/82552403>

Reviewed by NOBODY (OOPS!).

* committers-autocomplete.js: Use metadata/contributors.json.
(parseCommittersPy): Parse contributors.json as list instead of dictionary.
(isMatch): Strip irc references.
(updateMenu): Ditto.

JonWBedard committed:
[contributors.json] Relocation (Part 5)
https://bugs.webkit.org/show_bug.cgi?id=229690
<rdar://problem/82552403>

Reviewed by NOBODY (OOPS!).

* CISupport/ews-build/steps.py:
(ValidateCommiterAndReviewer): Pull from metadata/contributors.json.
(ValidateCommiterAndReviewer.load_contributors_from_disk): Ditto.
(ValidateCommiterAndReviewer.load_contributors): Parse contributors.json as list instead of dictionary.
* CISupport/ews-build/steps_unittest.py:

JonWBedard committed:
[contributors.json] Relocation (Part 6)
https://bugs.webkit.org/show_bug.cgi?id=229690
<rdar://problem/82552403>

Reviewed by NOBODY (OOPS!).

* WebKitBot/src/Commit.mjs: Load contributors from metadata/contributors.json.
* WebKitBot/src/Contributors.mjs: Load contributors from metadata/contributors.json, parse as list instead of dictionary.
* WebKitBot/tests/Commit.test.mjs: Use name instead of fullName, stop relying on nicks.
